sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -99,14 +77,83 @@ rom_error_t flash_ctrl_read_start(uint32_t addr, uint32_t word_count,
   return kErrorOk;
 }
 
-size_t flash_ctrl_fifo_read(size_t word_count, uint32_t *data_out) {
+void flash_ctrl_init(void) {
+  // Initialize the flash controller.
+  abs_mmio_write32(kBase + FLASH_CTRL_INIT_REG_OFFSET,
+                   bitfield_bit32_write(0u, FLASH_CTRL_INIT_VAL_BIT, true));
+}
+
+rom_error_t flash_ctrl_status_get(flash_ctrl_status_t *status) {
+  if (status == NULL) {
+    return kErrorFlashCtrlInvalidArgument;
+  }
+
+  // Read flash operation status.
+  uint32_t op_status = abs_mmio_read32(kBase + FLASH_CTRL_OP_STATUS_REG_OFFSET);
+  // Read flash controller status.
+  uint32_t fc_status = abs_mmio_read32(kBase + FLASH_CTRL_STATUS_REG_OFFSET);
+
+  // Extract operation status bits.
+  status->done = bitfield_bit32_read(op_status, FLASH_CTRL_OP_STATUS_DONE_BIT);
+  status->err = bitfield_bit32_read(op_status, FLASH_CTRL_OP_STATUS_ERR_BIT);
```

Comment:
Couldn't find where these bits are cleared (hwip tech spec page says they are rw). I think this function should clear these bits but we should use `~op_status` to avoid race conditions.

Reply:
Also wanted to point out that `OP_STATUS` bits are `rw` while `ERR_CODE` bits are `rw1c`.